metapackages/auto-instrumentations-node/src/utils.ts

```diff
@@ -136,3 +161,63 @@ export function getNodeAutoInstrumentations(
 
   return instrumentations;
 }
+
+export function getResourceDetectorsFromEnv(): Array<Detector | DetectorSync> {
+  const resourceDetectorsFromEnv =
+    process.env.OTEL_NODE_RESOURCE_DETECTORS?.split(',');
+
+  if (
+    resourceDetectorsFromEnv == null ||
+    resourceDetectorsFromEnv.includes('all')
+  ) {
+    return [
+      // Standard resource detectors.
+      containerDetector,
+      envDetectorSync,
+      hostDetectorSync,
+      osDetectorSync,
+      processDetectorSync,
+      // Cloud resource detectors.
+      alibabaCloudEcsDetector,
+      // Ordered AWS Resource Detectors as per:
+      // https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/processor/resourcedetectionprocessor/README.md#ordering
+      awsEksDetector,
+      awsEc2Detector,
+      gcpDetector,
+    ];
+  }
+
+  const resourceDetectors: Array<Detector | DetectorSync> = [];
+  for (const detector of resourceDetectorsFromEnv) {
```

Comment:
nit: the same logic [is applied by `BasicTracerProvider.ts`](https://github.com/open-telemetry/opentelemetry-js/blob/4b04407ba34594d5febc263fe6c79fd4cf7475d5/packages/opentelemetry-sdk-trace-base/src/BasicTracerProvider.ts#L276) on exporters and propagators environment configuration with a map :

```js
  protected _buildExporterFromEnv(): SpanExporter | undefined {
    const exporterName = getEnv().OTEL_TRACES_EXPORTER;
    if (exporterName === 'none' || exporterName === '') return;
    const exporter = this._getSpanExporter(exporterName);
    if (!exporter) {
      diag.error(
        `Exporter "${exporterName}" requested through environment variable is unavailable.`
      );
    }
    return exporter;
  }
```

```js
  protected static override _registeredExporters = new Map<
    string,
    () => SpanExporter
  >([
    ['otlp', () => this.configureOtlp()],
    ['zipkin', () => new ZipkinExporter()],
    ['jaeger', () => new JaegerExporter()],
    ['console', () => new ConsoleSpanExporter()],
  ]);

```


Could be nice to keep consistency and apply a similar pattern for detectors as well.

Reply:
Updated with a map as suggested. However, I had to create a map with the type `Map<string, any>` since the map contains `Detector`, `DetectorSync`, and an array of `Detector`.